pennylane/pauli/dla/lie_closure.py

```diff
@@ -422,12 +422,11 @@ def _check_independence(M, pauli_sentence, pw_to_idx, rank, num_pw, tol):
             M[pw_to_idx[pw], rank] = value
 
         # Check if new vector is linearly dependent on the current basis
-        v = M[:, -1].copy()  # remove copy to normalize M
+        v = M[:, -1]  # .copy()  # remove copy to normalize M
         v /= np.linalg.norm(v)
         A = M[:, :-1]
-        v = v - A @ qml.math.linalg.solve(qml.math.conj(A.T) @ A, A.conj().T) @ v
 
-        if np.linalg.norm(v) > tol:
+        if np.linalg.norm(A @ qml.math.linalg.pinv(A) @ v - v) > tol:
```

Comment:
Not sure what NumPy's associativity rules are, but we could do the following for good measure.
```suggestion
        if np.linalg.norm(A @ (qml.math.linalg.pinv(A) @ v) - v) > tol:
```

Reply:
I think it would actually first do `A @ pinv(A)`, but it should not matter. However, I like making this one clear and picking the cheaper order explicitly :)